db/R/query.format.vars.R

```diff
@@ -112,6 +112,22 @@ query.format.vars <- function(bety,input.id=NA,format.id=NA,var.ids=NA){
                    lat = site.lat,
                    lon = site.lon
     )
+    
+    # Check that all bety units are convertible. If not, throw a warning. 
+    for(i in 1:length(format$vars$bety_units)){
+      
+      if( format$vars$storage_type[i] != ""){ #units with storage type are a special case
+        
+        # This would be a good place to put a test for valid sotrage types. Currently not implemented. 
+        
+      }else if(udunits2::ud.are.convertible(format$vars$input_units[i], format$vars$pecan_units[i]) == FALSE){ 
+        
+        if(misc.are.convertible(format$vars$input_units[i], format$vars$pecan_units[i]) == FALSE){
```

Comment:
@istfer I'm not following you're request -- this already is checking if the units are convertable. If the units are convertable there's no need to check if they're the same, since that is just a special case of being convertable to itself.

As for the specific units listed, "ppm (= umol mol-1)", this is a great example of this function working as intended. That string does not represent valid units. Even though the string may be the same in both places, it's wrong in both places and needs to be fixed (rather than adding a special case that would let it through).